### PR TITLE
Aisrv integration and device testing

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
@BrennanGit , could you please check the Jenkins changes? Thanks!

File: Jenkinsfile [hunk 1]
```diff
@@ -24,99 +20,143 @@ pipeline {
...
-                sh "rm -rf *"
-                // clone
-                checkout([
-                    $class: 'GitSCM',
-                    branches: scm.branches,
-                    doGenerateSubmoduleConfigurations: false,
-                    extensions: [[$class: 'SubmoduleOption',
-                                  threads: 8,
-                                  timeout: 20,
-                                  shallow: false,
```
Comment (andrewstanfordjason) [inline Jenkinsfile:50]: why not shallow?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
         BAZEL_CACHE_URL = 'http://srv-bri-bld-cache:8080'
     }
     parameters { // Available to modify on the job page within Jenkins if starting a build
         string( // use to try different tools versions
             name: 'TOOLS_VERSION',
-            defaultValue: '15.0.6',
+            defaultValue: '15.2.1',
             description: 'The tools version to build with (check /projects/tools/ReleasesTools/)'
         )
         booleanParam( // use to check results of rolling all conda deps forward
```
Comment (BrennanGit) [inline Jenkinsfile]: I think you can remove the UPDATE_ALL param now as conda has gone.

File: Jenkinsfile (hunk 1 above)
Comment (andrewstanfordjason) [reply Jenkinsfile:50]: never mind! this is in the remove of the diff

File: Jenkinsfile [hunk 3]
```diff
@@ -24,99 +20,143 @@ pipeline {
...
+                                    sh "pytest integration_tests/runner.py --models_path integration_tests/models/bnns --bnn -n 8 --junitxml=integration_bnns_junit.xml"
+                                }
+                                // Any call to pytest can be given the "--junitxml SOMETHING_junit.xml" option
+                                // This step collects these files for display in Jenkins UI
+                                junit "**/*_junit.xml"
+                                // regression test for xmos_ai_tools juypiter notebooks
+                                withVenv {
+                                    sh "pip install pytest nbmake"
+                                    sh "pytest --nbmake ./docs/notebooks/*.ipynb"
+                                }
```
Comment (BrennanGit) [inline Jenkinsfile]: You can just wrap the whole thing in a single `withVenv`

```suggestion
                                withVenv {
                                    dir("experimental/xformer") {
                                        // xformer2 unit tests
                                        sh "./bazelisk-linux-amd64 test --remote_cache=${env.BAZEL_CACHE_URL} //Test:all --verbose_failures --test_output=errors --//:disable_version_check"
                                    }
                                    // xformer2 integration tests
                                    sh "pytest integration_tests/runner.py --models_path integration_tests/models/non-bnns -n 8 --junitxml=integration_non_bnns_junit.xml"
                                    sh "pytest integration_tests/runner.py --models_path integration_tests/models/bnns --bnn -n 8 --junitxml=integration_bnns_junit.xml"
                                    // Any call to pytest can be given the "--junitxml SOMETHING_junit.xml" option
                                    // This step collects these files for display in Jenkins UI
                                    junit "**/*_junit.xml"
                                    // regression test for xmos_ai_tools juypiter notebooks
                                    sh "pip install pytest nbmake"
                                    sh "pytest --nbmake ./docs/notebooks/*.ipynb"
                                }
```

**Review by andrewstanfordjason — APPROVED** [review]

**Review by BrennanGit — APPROVED** [review]
Jenkinsfile changes look good, a couple of cleanup options maybe. Functionally seems to work as expected.

File: Jenkinsfile (hunk 1 above)
Comment (ghost) [reply Jenkinsfile:50]: That's fine, no worries. shallow does not checkout tags which we need for version checks and to make python wheel.

**Pull request merged**